libs/filamat/src/shaders/CodeGenerator.cpp

```diff
@@ -94,6 +111,17 @@ utils::io::sstream& CodeGenerator::generateProlog(utils::io::sstream& out, Shade
     // #included code. This way, glslang reports errors more accurately.
     out << "#extension GL_GOOGLE_cpp_style_line_directive : enable\n\n";
 
+    if (v.hasStereo() && stage == ShaderStage::VERTEX) {
+        switch (material.stereoscopicType) {
+        case StereoscopicType::INSTANCED:
+            // Nothing to generate
+            break;
+        case StereoscopicType::MULTIVIEW:
+            out << "layout(num_views = " << material.stereoscopicEyeCount << ") in;\n";
```

Comment:
shouldn't this be `CONFIG_STEREO_EYE_COUNT` ?

Reply:
No, it should be the number of currently configured number.

Reply:
but then how is the spec constant used in the shader?

Reply:
That spec constant, CONFIG_STEREO_EYE_COUNT, is only being referenced for instanced stereo. And this code generation is only for multiview stereo.

Reply:
So it's *not* possible to change the number of views dynamically via spec constant?

Reply:
hmm, is that a limitation of glsl? (where you can't put a `#define` constant in `layout()` or something)?  @pixelflinger , would you mind weighing in on your original comment concerning spec const?

Reply:
AFAICT, this is where spec constant is generated https://github.com/google/filament/blob/d21613a5e666e5e28280d8f19b9a191d23453083/libs/filamat/src/shaders/CodeGenerator.cpp#L862-L879

And this is where the constant is added. https://github.com/google/filament/blob/d21613a5e666e5e28280d8f19b9a191d23453083/libs/filamat/src/shaders/CodeGenerator.cpp#L295

In the function, it uses #define for non-spirv, in which case it is possible to synthesize something like "layout(view_nums=MY_DEFINE)". However, it's not possible in the other case like "layout(view_nums=g_eyecnt_variable)".

There are many places where it directly generates "layout" in CodeGenerator.cpp. One example is https://github.com/google/filament/blob/d21613a5e666e5e28280d8f19b9a191d23453083/libs/filamat/src/shaders/CodeGenerator.cpp#L98

Reply:
I see so the problem is the spirv codegen.  In that case, I would just generate `"layout(num_views = " << material.stereoscopicEyeCount << ")` for spirv and use the `#define` for gl/metal. Since the vk backend will have to do a non-trivial substitution for this regardless.